src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -136114,27 +136536,31 @@ static void F<T>(Func<T> f)
     {
         F(() =>
         {
-            if (b) return default; // 1
+            if (b) return default;
```

Comment:
Does your change fix https://github.com/dotnet/roslyn/issues/44339? If so, consider adding a WorkItem to this test linking to that issue, and consider adding a test for the specific in the issue.

Also consider checking if this change resolves https://github.com/dotnet/roslyn/issues/43536.